php/src/Communicator.cpp

```diff
@@ -911,6 +911,33 @@ ZEND_FUNCTION(Ice_initialize)
         initData.properties = std::make_shared<Ice::Properties>();
     }
 
+    if (initData.properties->getIceProperty("Ice.ProgramName").empty())
+    {
+        // Try to get the script filename from PHP's global variables
+        zval* scriptName = zend_hash_str_find(&EG(symbol_table), "_SERVER", sizeof("_SERVER") - 1);
```

Comment:
`scriptName` doesn't seems a good name for this variable, this is the array containing all entries with info about the execution environment. What about `scriptInfo`